### PR TITLE
Snow add versioning argument

### PR DESCRIPTION
**Pull request by anas-yousef**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-32666

## Description
Added the argument *force_default_url* to the command *servicenow-create-co-from-template*, which will ignore the api version configured in the instance configuration when set to true.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 1]
```diff
@@ -53,6 +55,51 @@ def util_load_json(path):
...
+        demisto,
+        'params',
+        return_value={
+            'isFetch': True,
+            'url': url,
+            'credentials': {
+                'identifier': 'identifier',
+                'password': 'password',
+            },
+            'api_version': api_version,
```
Comment (dorschw) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:79]:
```suggestion
    mocker.patch.object(
        demisto,
        'params',
        return_value={
            'isFetch': True,
            'url': url,
            'credentials': {
                'identifier': 'identifier',
                'password': 'password',
            },
            'api_version': 'v2', # << We test overriding this value
```

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -3005,7 +3006,10 @@ def main():
         password = params.get('credentials', {}).get('password')
 
     version = params.get('api_version')
-    if version:
+    # This is given as part of the arguments of the command, if True, then the request will not the configured api version
+    force_default_url = argToBoolean(args.get('force_default_url', 'false'))
```
Comment (dorschw) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:3010]: mention in the comment that it only exists in specific commands

**Review by dorschw — COMMENTED** [review]
Very nice

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 3]
```diff
@@ -53,6 +55,51 @@ def util_load_json(path):
...
+    mocker.patch.object(
+        demisto,
+        'args',
+        return_value={
+            'template': 'dummy_template',
+            'force_default_url': 'true'
+        }
+    )
+    mocker.patch.object(demisto, 'command', return_value='servicenow-create-co-from-template')
+    requests_mock.post(ANY, json=util_load_json('test_data/create_co_from_template_result.json'))
```
[the remark this answers is not in the document]
Comment (anas-yousef) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:98]: Fixed it

**Comment by anas-yousef** [comment]
This will need a force merge, since the step **xsoar_ng_server_ga** keeps on failing from multiple internal machine issues, such as:
1. **failed to get all integrations configuration**
2. **Failed to create incident**
3. **Integration not Found**
@yuvalbenshalom

**Comment by anas-yousef** [comment]
@yuvalbenshalom please force-merge, reason [here](https://panw-global.slack.com/archives/G011E63JXPB/p1707054352345729), [build](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/jobs/4585311)

**Pull request merged**